### ISSUE TITLE
Trezor support on Windows versions

### ISSUE THREAD

**Issue by RustyNomad** [dashpay/electrum-dash#21]
Tried both the Windows-Setup and Windows (stand alone) versions and received same error.

When setting up a new Trezor (hardware wallet) following message comes up:

Trezor device not detected.
Continuing in watching-only mode.
Reason: please install github.com/trezor/python-trezor

See same error was also reported by other windows users on dash forum.

**Comment by schinzelh** [comment]
Please try this version

https://dashpay.atlassian.net/builds/browse/ELMW-DEV-117/artifact/JOB1/electrum-dash.exe/electrum-dash.exe

**Comment by RustyNomad** [comment]
@schinzelh Not working, when started the dialog windows open with an error about the config file that could not be found. It does however startup after that and I'm able to go through the steps to setup a Trezor wallet but once I click on OK on the password window (after the pin) Electrum just closes with nothing further happening.

**Comment by schinzelh** [comment]
The message complaining about the missing config file is normal.

What Firmware version does your Trezor have?

**Comment by RustyNomad** [comment]
@schinzelh The latest update 1.4.0

**Comment by schinzelh** [comment]
Yeah, it seems 1.4.0 broke the USB/HID code. I'll upgrade my as well and see if i can reproduce/fix the issue. I'll come back to you tomorrow.

**Comment by schinzelh** [comment]
I have updated my Trezor to 1.4.0 - funny enough it still works with the build above, though it's acting a bit weird as Electrum is show two(!) devices i can choose from.

![image](https://cloud.githubusercontent.com/assets/636669/18582863/87ffa288-7c07-11e6-9a79-a4aff1f4ba4b.png)


One works, one does not :)

**Comment by dsaki** [comment]
I have same problem on Electrun-Dash OSX.
![2016-09-18 13 36 36](https://cloud.githubusercontent.com/assets/1742236/18612917/14078162-7da5-11e6-9966-b061292faae1.png)

And I tryed build wallet like above, but there is one choice to select device...

<img width="577" alt="2016-09-18 13 39 14" src="https://cloud.githubusercontent.com/assets/1742236/18612928/6735ddb6-7da5-11e6-8ef3-7514ce405ebe.png">

My trezor version is 1.4.0. Please help me.

**Comment by SpaceAgeMinds** [comment]
I can confirm that Electrum-Dash 2.4.1 does not work with either python-trezor 0.7.0 and python-trezor 0.6.4 (I thought maybe an older version might work). I installed python-trezor using pip and used the standard Electrum-Dash setup executable.

I'm running Windows 10, Trezor firmware is 1.4.0, I also have the newest Trezor bridge installed if that helps.

On an a maybe unrelated note It also doesn't work on Debain Linux but for a different reason (mushed window). I'm not currently sitting at that work station so I can't give you more specifics.

**Comment by tgzcode** [comment]
I think that current problem here is two-part (It was for me..)
Thanks to Satoshilabs support for all their great help and to member schinzelh who showed up here the day I pointed out to Satosilabs that RustyNomad had my same problem way back since April.
My TREZOR hardware wallet is now working (older version electrum-DASH) - it does load with a choice between two devices that are both my same device plugged in - I just use the top one on list which is selected default & continue. Hope this helps anyone or maybe with debugging latest electrum-DASH wallet.

My set-up:
OS is Windows 7 Professional - Service Pack 1
trezor-bridge-1.2.0-win32-
latest TREZOR Firmware update 1.4.0 (THIS BROKE WIN DEVICE DRIVER)
---Problem-----
I downloaded & ran latest bridge program - trezor-bridge-1.2.0-win32-install
I then went to mytrezor .com and plugged my trezor in.
Account reported good data but there was a firmware upgrade notice for 1.4.0.
I attempted to upgrade firmware which was downloaded and installed.
after unplugging trezor - reloading page - and re-connecting trezor - the website never responds to my device.

My CPU signals a sound for new device added when I plug trezor into USB but the website never updates, displaying message to "Get started! connect your TREZOR device to the computer." 
---Fix----
My Windows device manager reports a problem under the following:
 -Human Interface Devices
 --USB Input Device
 The Device cannot start (Code 10)
(I've seen in forums a step to click [troubleshoot driver problem] but my OS does not show that option and I did not know where the correct driver was located for installation)
Sooo From device manager clicked [update driver]..
I selected [browse my computer for driver software] then [Let me pick from a list of device drivers on my computer].
Next there were instructions to select manufacturer and model of device but only one choice was available:
USB Composite Device.
It was selected, I clicked [Next] and driver updated/changed to following:
C:\Windows\system32\drivers\usbccgp.sys
(previously listed 3 .dll files - I didn't record names..)
My TREZOR was again working at MyTREZOR site and Satoshilabs support approved my fix.

Next I tried to install electrum-DASH using Electrum Dash (2.4.1) windows 32.64 installer.
All proceeded normal until I opened electrum-DASH wallet and tried to create a hardware wallet.
---Problem----
After selecting the TREZOR device and [Next] - I received and error:
error device not detected
continuing in watching-only mode.
Reason:
please install github.com/trezor/python-trezor [OK]

Satoshilabs had just helped with above driver problem and I asked them about this.
They stated that they can't solve problems with third-party products, our developers need to integrate Satoshilabs latest features when they roll out -but- they said as a work-around solution of an older version of electrum-DASH would let it run in full mode.
---The Fix----
It is the same file given above by member schinzelh, the file in this directory:
https://dashpay.atlassian.net/builds/browse/ELMW-DEV-117/artifact/JOB1/electrum-dash.exe/

I've been running since loading this version, using the TREZOR for DASH and also with myTREZOR web wallet for BTC. Will be checking back here to see about any updates to a new patched electrum-DASH.

**Comment by Kefkius** [comment]
Electrum-DASH currently isn't using TREZOR Bridge. Electrum's non-release code seems to have this, but it's currently [disabled](https://github.com/spesmilo/electrum/blob/master/plugins/trezor/plugin.py#L108).

> `# disable bridge because it seems to never returns if keepkey is plugged`

@schinzelh I can make a PR that attempts to use TREZOR Bridge in addition to the usual HID. It might fix the problem in cases where mytrezor.com works but Electrum-DASH doesn't.

**Comment by SpaceAgeMinds** [comment]
Just tried the version listed in @tgzcode post. It does not work under Windows 10. It's a lot closer, even getting me far as entering my P.I.N but something different happened each time and it would not work. Frozen Trezor, weird blank pop up, or just flat out crashing.

If anyone is able to get it working under Windows 10 please post steps.

**Comment by tgzcode** [comment]
@SpaceAgeMinds What does your Win 10 use as a driver for the TREZOR?
Check windows device manager with the TREZOR plugged in.
For me it was not grouped with the other USB drivers but a "New" Listing named "Human Interface Devices". and it was flagged with a problem after TREZOR update to 1.4.0.

The pop-up I get when loading is one that reads the name that I've given the TREZOR and asks me to select it. (and weirdly lists the same device again - I've only got one TREZOR so can't test this with two plugged in, where some selection would make more sense.)

@Kefkius points out that the TREZOR bridge is not needed but using it for BTC wallet did alert me to the win driver problem. Maybe he knows if the electrum-DASH python code requires any working win drivers or if it works around that?

**Comment by SpaceAgeMinds** [comment]
@tgzcode nothing shows up in my device manager.

**Comment by tgzcode** [comment]
@SpaceAgeMinds OK, tested some things here:
Plugged in TREZOR with bridge software not loaded, (trezord.exe not running) and device manager still shows TREZOR connection. With device manager open I plug TREZOR into USB. There is a windows sound for device connection and TREZOR lights up showing it's name that I gave it. Windows Device Manager refreshes with an added listing of "Human Interface Devices".

If your TREZOR remains dead, you might try using any other available USB ports. USB Ports can be powered or non-powered but there should be a main powered USB on your system. TREZOR must need power to light it's screen...

**Comment by Kefkius** [comment]
I'm a little confused, @tgzcode @SpaceAgeMinds are you trying the build that was made after #84?

**Comment by SpaceAgeMinds** [comment]
@Kefkius I tried using the build @tgzcode has linked in his post on sept. 19th on this page. I have not tried building the code from #84, do you have Windows 10 build instructions?
